app/src/main/java/org/mozilla/fenix/compose/HorizontalFadingEdge.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+
+/**
+ * Displays the [content] with the right edge fading.
+ *
+ * @param modifier [Modifier] for the container.
+ * @param fadeWidth Length of the fading edge.
+ * @param backgroundColor Color of the background shown under the content.
+ * @param isContentRtl Whether or not the content should be displayed Right to Left
+ * @param content The content whose right edge must be faded.
+ */
+@Composable
+fun HorizontalFadingEdge(
+    modifier: Modifier = Modifier,
+    fadeWidth: Dp = 25.dp,
+    backgroundColor: Color = Color.Transparent,
+    isContentRtl: Boolean = false,
+    content: @Composable () -> Unit
```

Comment:
```suggestion
    content: @Composable BoxScope.() -> Unit
```

This allows the passed-in `content` to use any `Modifier` directly applicable to a `BoxScope`